librad/src/net/protocol/membership/hpv.rs

```diff
@@ -124,20 +123,18 @@ where
     Addr: Clone + Debug + PartialEq,
 {
     pub fn new(
-        spawner: &executor::Spawner,
         local_id: PeerId,
         rng: Rng,
         params: Params,
-    ) -> (Self, mpsc::Receiver<Periodic<Addr>>)
+    ) -> (Self, BoxStream<'static, Periodic<Addr>>)
```

Comment:
Nit: can we defer the boxing until absolutely needed (ie. return `impl Stream` here)?